src/WebSocketHandler.js

```diff
@@ -5,246 +5,35 @@
-
-    // Store variable that is passed to Redux if ws is online
-    this.isOnline = undefined;
-
-    // Heartbeat interval in milliseconds.
-    this.heartbeatInterval = HEARTBEAT_TMO;
-
-    // Connection timeout.
-    this.connectionTimeout = 5000;
-
-    // Retry connection interval in milliseconds.
-    this.retryConnectionInterval = 1000;
-
-    // Open connection timeout.
-    this.openConnectionTimeout = 20000;
-
-    // Date of latest setup call. The setup is the way to open a new connection.
-    this.latestSetupDate = null;
-
-    // Date of latest ping.
-    this.latestPingDate = null;
-
-    // Latest round trip time measured by PING/PONG.
-    this.latestRTT = null;
-
-    // Timer used to detected when connection is down.
-    this.timeoutTimer = null;
-  }
-
-  /**
-   * Start websocket object and its methods
-   */
-  setup() {
-    if (this.started) {
-      return;
-    }
-    let wsURL = helpers.getWSServerURL();
-    if (wsURL === null) {
-      return;
-    }
-
-    if (this.ws) {
-      const dt = (new Date() - this.latestSetupDate) / 1000;
-      if (dt < this.openConnectionTimeout) {
-        return;
-      }
-      this.ws.close();
-    }
-    this.ws = new this.WebSocket(wsURL);
-    this.latestSetupDate = new Date();
-
-    this.ws.onopen = () => {
-      this.onOpen();
-    }
-    this.ws.onmessage = (evt) => {
-      this.onMessage(evt);
-    }
-    this.ws.onerror = (evt) => {
-      this.onError(evt);
-    }
-    this.ws.onclose = () => {
-      this.onClose();
-    }
-  }
-
-  onPong() {
-    if (this.latestPingDate) {
-      const dt = (new Date() - this.latestPingDate) / 1000;
-      this.latestRTT = dt;
-      this.latestPingDate = null;
-    }
-    if (this.timeoutTimer) {
-      clearTimeout(this.timeoutTimer);
-      this.timeoutTimer = null;
-    }
-  }
-
-  /**
-   * Handle message receiving from websocket
-   *
-   * @param {Object} evt Event that has data (evt.data) sent in the websocket
-   */
-  onMessage(evt) {
-    const message = JSON.parse(evt.data)
-    const _type = message.type.split(':')[0]
-    if (_type === 'pong') {
-      this.onPong();
-    }
-    this.emit(_type, message)
-  }
-
-  /**
-   * Method called when websocket connection is opened
-   */
-  onOpen() {
-    if (this.connected === false) {
-      // If was not connected  we need to reload data
-      // Emits event to reload data
-      this.emit('reload_data');
-    }
-    this.connected = true;
-    this.started = true;
-    this.setIsOnline(true);
-    this.heartbeat = setInterval(() => {
-      this.sendPing();
-    }, this.heartbeatInterval);
-    wallet.onWebsocketOpened();
-  }
-
-  /**
-   * Method called when websocket connection is closed
-   */
-  onClose() {
-    this.started = false;
-    this.connected = false;
-    this.setIsOnline(false);
-    wallet.onWebsocketBeforeClose();
-    if (this.ws) {
-      this.ws.close();
-      this.ws = null;
-    }
+import helpers from './helpers';
+import WS from './websocket';
```

Comment:
What is the goal of having `WebSocketHandler.js` and `websocket.js`? Is it for keeping the interface?

If that's the case, this should still be an event emitter and we should emit all events. For instance, this is a snippet from our wallet code:
```js
hathorLib.WebSocketHandler.on('height_updated', this.handleHeightUpdated);
```

This won't work after this change. I'm not saying that we should keep compatibility. Actually, we could discuss it. Maybe it's easier to break it already, as I don't believe other people are using the lib.

Reply:
Done that, can you see what you think?

Reply:
Looks good.